modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/DataTypesTest.java

```diff
@@ -32,6 +33,77 @@
  * Test SQL data types.
  */
 public class DataTypesTest extends AbstractBasicIntegrationTest {
+    /** Tests UUID without index. */
+    @Test
+    public void testUUIDWithoutIndex() {
+        testUUID(false);
+    }
+
+    /** Tests UUID with the index. */
+    @Test
+    public void testUUIDWithIndex() {
+        testUUID(true);
+    }
+
+    /** Tests UUID type. */
+    private void testUUID(boolean indexed) {
+        try {
+            executeSql("CREATE TABLE t(id INT, name VARCHAR(255), uid UUID, primary key (id))");
+
+            if (indexed)
+                executeSql("CREATE INDEX uuid_idx ON t (uid);");
+
+            executeSql("INSERT INTO t VALUES (1, 'fd10556e-fc27-4a99-b5e4-89b8344cb3ce', " +
+                "'9e120341-627f-32be-8393-58b5d655b751')");
+            // Name == UUID
+            executeSql("INSERT INTO t VALUES (2, '123e4567-e89b-12d3-a456-426614174000', " +
+                "'123e4567-e89b-12d3-a456-426614174000')");
+
+            assertQuery("SELECT * FROM t")
+                .returns(1, "fd10556e-fc27-4a99-b5e4-89b8344cb3ce", UUID.fromString("9e120341-627f-32be-8393-58b5d655b751"))
+                .returns(2, "123e4567-e89b-12d3-a456-426614174000", UUID.fromString("123e4567-e89b-12d3-a456-426614174000"))
+                .check();
+
+            assertQuery("SELECT * FROM t WHERE uid < '123e4567-e89b-12d3-a456-426614174000'")
+                .returns(1, "fd10556e-fc27-4a99-b5e4-89b8344cb3ce", UUID.fromString("9e120341-627f-32be-8393-58b5d655b751"))
+                .check();
+
+            assertQuery("SELECT * FROM t WHERE '123e4567-e89b-12d3-a456-426614174000' > uid")
+                .returns(1, "fd10556e-fc27-4a99-b5e4-89b8344cb3ce", UUID.fromString("9e120341-627f-32be-8393-58b5d655b751"))
+                .check();
+
+            assertQuery("SELECT * FROM t WHERE name = uid")
+                .returns(2, "123e4567-e89b-12d3-a456-426614174000", UUID.fromString("123e4567-e89b-12d3-a456-426614174000"))
+                .check();
+
+            assertQuery("SELECT * FROM t WHERE name != uid")
+                .returns(1, "fd10556e-fc27-4a99-b5e4-89b8344cb3ce", UUID.fromString("9e120341-627f-32be-8393-58b5d655b751"))
+                .check();
+
+            assertQuery("SELECT count(*), uid FROM t group by uid order by uid")
```

Comment:
Please add also min/max aggregation test (`SELECT min(uid), max(uid) FROM t`). `Accumulators` class should be fixed, see `VarBinaryMinMax` as an example (perhaps we should generalize `VarBinaryMinMax` to avoid boilerplate code).

Reply:
SUM and AVG is confusing for UUID, I think it's better to throw error like 'not supported' (the same as for VARCHAR type)

Reply:
Fixed